File: 0735-asteroid-collision/0735-asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] A) {
        Stack<Integer> st = new Stack<>();
        for (int a : A) {
            int flag = 1;
            while (!st.empty() && (st.peek() > 0 && a < 0)) {
                if (Math.abs(a) > Math.abs(st.peek())) {
                    st.pop();
                    continue;
                }
                if (Math.abs(a) == Math.abs(st.peek())) {
                    st.pop();
                }
                flag = 0;
                break;
            }
            if (flag == 1) {
                st.push(a);
            }
        }
        int[] ans = new int[st.size()];
        for (int i = st.size() - 1; i >= 0; --i) {
            ans[i] = st.peek();
            st.pop();
        }
        return ans;
    }
}

File: 0735-asteroid-collision/0735-asteroid-collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& A) {
        stack<int> st;
        for(int a : A) {
            int flag = 1;
            while(!st.empty() && (st.top() > 0 && a < 0)) {
                if(abs(a) > abs(st.top())) {
                    st.pop(); continue;
                }
                if(abs(a) == abs(st.top())) {
                    st.pop();
                }
                flag = 0;
                break;
            }
            if(flag) st.push(a);
        }
        
        vector<int> ans(st.size());
        for(int i = st.size() - 1; i >= 0; --i) {
            ans[i] = st.top(); st.pop();
        }
        
        return ans;
    }
};